### ISSUE TITLE
Select depend on another select, ajax 

### ISSUE THREAD

**Issue by milos-devme** [Laravel-Backpack/CRUD#2952]
# Bug report
Hello, this is my first time using laravel backpack and it is great! I love it, and I just start using it. I already installed it on my existing project but I need help about select that depends on another select. I followed instructions in docs but still nothing happens.
I'm using ```Laravel 5.8``` and ```Backpack 3.6``` versions.
### What I did
I followed instructions on Backpack docs, for Add a select2 field that depends on another field on this link. So what I did first:

I created two tables, Category table and Article table. Their models:
```
class Category extends Model
{
    use CrudTrait;

    /*
    |--------------------------------------------------------------------------
    | GLOBAL VARIABLES
    |--------------------------------------------------------------------------
    */

    protected $table = 'categories';
    protected $primaryKey = 'id';
    // public $timestamps = false;
    // protected $guarded = ['id'];
    protected $fillable = ['title'];
    // protected $hidden = [];
    // protected $dates = [];

    /*
    |--------------------------------------------------------------------------
    | FUNCTIONS
    |--------------------------------------------------------------------------
    */

    public function articles(){
        return $this->hasMany('App\Models\Article');
    }
}
```
And Article model:
```
class Article extends Model
{
    use CrudTrait;

    /*
    |--------------------------------------------------------------------------
    | GLOBAL VARIABLES
    |--------------------------------------------------------------------------
    */

    protected $table = 'articles';
    protected $primaryKey = 'id';
    // public $timestamps = false;
    // protected $guarded = ['id'];
    protected $fillable = ['title', 'category_id'];
    // protected $hidden = [];
    // protected $dates = [];

    /*
    |--------------------------------------------------------------------------
    | FUNCTIONS
    |--------------------------------------------------------------------------
    */

    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    |--------------------------------------------------------------------------
    */

    public function category(){
        return $this->belongsTo('App\Models\Category');
    }
}
```
I made relationships between these two models like this. So when I create Article it shows me title and I have to choose category_id from category select.

After all this I made Archive table and these are my migrations:
```
Schema::create('archives', function (Blueprint $table) {
   $table->increments('id');
   $table->string('title');

   $table->bigInteger('category_id')->unsigned();
   $table->foreign('category_id')->references('id')->on('categories');

   $table->bigInteger('article_id')->unsigned();
   $table->foreign('article_id')->references('id')->on('articles');

   $table->timestamps();
});
```
And my Archive.php model:
```
class Archive extends Model
{
    use CrudTrait;

    /*
    |--------------------------------------------------------------------------
    | GLOBAL VARIABLES
    |--------------------------------------------------------------------------
    */

    protected $table = 'archives';
    protected $primaryKey = 'id';
    // public $timestamps = false;
    // protected $guarded = ['id'];
    protected $fillable = ['title', 'category', 'article'];
    // protected $hidden = [];
    // protected $dates = [];

    /*
    |--------------------------------------------------------------------------
    | FUNCTIONS
    |--------------------------------------------------------------------------
    */

    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    |--------------------------------------------------------------------------
    */

    public function category(){
        return $this->belongsTo('App\Models\Category');
    }

    public function article(){
        return $this->belongsTo('App\Models\Article');
    }
}
```

And then I followed instructions from the backpack docs. This is my ArchiveCrudController:

```
public function setup()
    {
        /*
        |--------------------------------------------------------------------------
        | CrudPanel Basic Information
        |--------------------------------------------------------------------------
        */
        $this->crud->setModel('App\Models\Archive');
        $this->crud->setRoute(config('backpack.base.route_prefix') . '/archive');
        $this->crud->setEntityNameStrings('archive', 'archives');

        $this->crud->setColumns(['title', 'category', 'article']);
        $this->crud->addField([
            'name' => 'title',
            'type' => 'text',
            'label' => "Archive title"
        ]);

        $this->crud->addField([    // SELECT2
            'label'         => 'Category',
            'type'          => 'select',
            'name'          => 'category_id',
            'entity'        => 'category',
            'attribute'     => 'title',
        ]);
        $this->crud->addField([ // select2_from_ajax: 1-n relationship
            'label'                => "Article", // Table column heading
            'type'                 => 'select2_from_ajax',
            'name'                 => 'article_id', // the column that contains the ID of that connected entity;
            'entity'               => 'article', // the method that defines the relationship in your Model
            'attribute'            => 'title', // foreign key attribute that is shown to user
            'data_source'          => url('api/article'), // url to controller search function (with /{id} should return model)
            'placeholder'          => 'Select an article', // placeholder for the select
            'minimum_input_length' => 0, // minimum characters to type before querying results
            'dependencies'         => ['category'], // when a dependency changes, this select2 is reset to null
            //'method'                    => ‘GET’, // optional - HTTP method to use for the AJAX call (GET, POST)
        ]);


        /*
        |--------------------------------------------------------------------------
        | CrudPanel Configuration
        |--------------------------------------------------------------------------
        */

        // TODO: remove setFromDb() and manually define Fields and Columns
        //$this->crud->setFromDb();

        // add asterisk for fields that are required in ArchiveRequest
        $this->crud->setRequiredFields(StoreRequest::class, 'create');
        $this->crud->setRequiredFields(UpdateRequest::class, 'edit');
    }
```
Like from backpack docs. Then I made Api folder in App\Http\Controller and then I made ArticleController in it, like this:

```

<?php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\Article;
use Illuminate\Http\Request;

class ArticleController extends Controller
{
    public function index(Request $request)
    {
        $search_term = $request->input('q');
        $form = collect($request->input('form'))->pluck('value', 'name');

        $options = Article::query();

        // if no category has been selected, show no options
        if (! $form['category']) {
            return [];
        }

        // if a category has been selected, only show articles in that category
        if ($form['category']) {
            $options = $options->where('category_id', $form['category']);
        }

        if ($search_term) {
            $results = $options->where('title', 'LIKE', '%'.$search_term.'%')->paginate(10);
        } else {
            $results = $options->paginate(10);
        }

        return $options->paginate(10);
    }

    public function show($id)
    {
        return Article::find($id);
    }
}
```
I just copy-paste code from docs but of course I changed model for my need. And finally my routes:
```
Route::get('api/article', 'App\Http\Controllers\Api\ArticleController@index');
Route::get('api/article/{id}', 'App\Http\Controllers\Api\ArticleController@show');
```
I copy-paste these routes in my web.php in routes folder and in custom.php in routes/backpack folder.

### What I expected to happen

I expected to show article posts from selected category.

### What happened

But when I choose category in my Archive create no articles showing. Can someone please help me with this?
![backpackadmin](https://user-images.githubusercontent.com/66961342/85129719-e7c8e380-b233-11ea-8f60-1a39de0c15ad.png)

**Comment by milos-devme** [comment]
I have my ```Stackoverflow``` question on [https://stackoverflow.com/questions/62468799/laravel-backpack-select-dependent-from-another-select](url)

**Comment by pxpm** [comment]
Hello @milos-devme if you are just starting with backpack i would suggest you to start with the latest backpack version. 

We are no longer pushing fixes or features to 3.6 version for sometime now.

Check out the new version and the updated docs: https://backpackforlaravel.com/docs/4.1/installation

Also, it would help if you provide the concrete error. You can get it on your `Network` tab on the developer tools in the browser when the ajax request is made if that is the case otherwise it should be in laravel logs or console in case it's a javascript error.

Not much we can help here, you did the best that is asking on Stackoverflow, I am sure someone will pitch in and help you with, but like I said, the error message is the best way to debug what is wrong with backpack or your code so I suggest you update your question with that information.

There are other mediums you can also ask there, I will let the bot close this one so you get the links. 

Wish you the best,
Pedro
